circleci: failed tests (13559): TestStoreRangeMergeStats

The following test appears to have failed:

[#13559](https://circleci.com/gh/cockroachdb/cockroach/13559):

```
I160301 01:04:01.755283 kv/range_cache.go:189  adding descriptor: key=/Meta2/"b" desc=range_id:1 start_key:"" end_key:"b" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I160301 01:04:01.755907 kv/range_cache.go:128  lookup range descriptor: key="b"
I160301 01:04:01.756073 kv/range_cache.go:189  adding descriptor: key=/Meta2/Max desc=range_id:2 start_key:"b" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I160301 01:04:01.756981 storage/replica_command.go:467  auto-gc'ed 1f7929b3 (8 intents)
I160301 01:04:01.765979 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- FAIL: TestStoreRangeMergeStats (0.18s)
	client_merge_test.go:413: failed to verify range's stats after merge: expected range's stats to agree with recomputation: got
		{LastUpdateNanos:0 IntentAge:0 GCBytesAge:0 LiveBytes:41470 LiveCount:219 KeyBytes:16289 KeyCount:220 ValBytes:25370 ValCount:225 IntentBytes:0 IntentCount:0 SysBytes:596 SysCount:8}
		recomputed
		{LastUpdateNanos:0 IntentAge:0 GCBytesAge:0 LiveBytes:41470 LiveCount:219 KeyBytes:16289 KeyCount:220 ValBytes:25370 ValCount:225 IntentBytes:0 IntentCount:0 SysBytes:780 SysCount:10}
=== RUN   TestStoreStatsSplit
I160301 01:04:01.769263 raft/raft.go:459  [group 1] 1 became follower at term 5
I160301 01:04:01.769400 raft/raft.go:235  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I160301 01:04:01.769430 raft/raft.go:539  [group 1] 1 is starting a new election at term 5
I160301 01:04:01.769465 raft/raft.go:472  [group 1] 1 became candidate at term 6
I160301 01:04:01.769490 raft/raft.go:521  [group 1] 1 received vote from 1 at term 6
--
I160301 01:04:31.775792 stopper.go:273  draining; tasks left:
41     gossip/infostore.go:285
I160301 01:04:31.775823 stopper.go:273  draining; tasks left:
31     gossip/infostore.go:285
--- PASS: Example_rebalancing (0.12s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	34.767s
=== RUN   TestBatchBasics
I160301 01:03:54.823706 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.02s)
=== RUN   TestBatchGet
I160301 01:03:54.840908 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.01s)
=== RUN   TestBatchMerge
I160301 01:03:54.846221 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.01s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/storage/client_merge_test.go
+++ b/storage/client_merge_test.go
@@ -380,18 +380,20 @@ func TestStoreRangeMergeStats(t *testing.T) {
 
 	// Get the range stats for both ranges now that we have data.
 	var msA, msB engine.MVCCStats
-	if err := engine.MVCCGetRangeStats(store.Engine(), aDesc.RangeID, &msA); err != nil {
+	snap := store.Engine().NewSnapshot()
+	defer snap.Close()
+	if err := engine.MVCCGetRangeStats(snap, aDesc.RangeID, &msA); err != nil {
 		t.Fatal(err)
 	}
-	if err := engine.MVCCGetRangeStats(store.Engine(), bDesc.RangeID, &msB); err != nil {
+	if err := engine.MVCCGetRangeStats(snap, bDesc.RangeID, &msB); err != nil {
 		t.Fatal(err)
 	}
 
 	// Stats should agree with recomputation.
-	if err := verifyRecomputedStats(store, aDesc, msA); err != nil {
+	if err := verifyRecomputedStats(snap, aDesc, msA); err != nil {
 		t.Fatalf("failed to verify range A's stats before split: %v", err)
 	}
-	if err := verifyRecomputedStats(store, bDesc, msB); err != nil {
+	if err := verifyRecomputedStats(snap, bDesc, msB); err != nil {
 		t.Fatalf("failed to verify range B's stats before split: %v", err)
 	}
 
@@ -403,14 +405,16 @@ func TestStoreRangeMergeStats(t *testing.T) {
 	rngMerged := store.LookupReplica(aDesc.StartKey, nil)
 
 	// Get the range stats for the merged range and verify.
+	snap = store.Engine().NewSnapshot()
+	defer snap.Close()
 	var msMerged engine.MVCCStats
-	if err := engine.MVCCGetRangeStats(store.Engine(), rngMerged.RangeID, &msMerged); err != nil {
+	if err := engine.MVCCGetRangeStats(snap, rngMerged.RangeID, &msMerged); err != nil {
 		t.Fatal(err)
 	}
 
 	// Merged stats should agree with recomputation.
-	if err := verifyRecomputedStats(store, rngMerged.Desc(), msMerged); err != nil {
-		t.Fatalf("failed to verify range's stats after merge: %v", err)
+	if err := verifyRecomputedStats(snap, rngMerged.Desc(), msMerged); err != nil {
+		t.Errorf("failed to verify range's stats after merge: %v", err)
 	}
 }
 

--- a/storage/client_split_test.go
+++ b/storage/client_split_test.go
@@ -413,11 +413,13 @@ func TestStoreRangeSplitStats(t *testing.T) {
 	writeRandomDataToRange(t, store, rng.RangeID, keyPrefix)
 
 	// Get the range stats now that we have data.
+	snap := store.Engine().NewSnapshot()
+	defer snap.Close()
 	var ms engine.MVCCStats
-	if err := engine.MVCCGetRangeStats(store.Engine(), rng.RangeID, &ms); err != nil {
+	if err := engine.MVCCGetRangeStats(snap, rng.RangeID, &ms); err != nil {
 		t.Fatal(err)
 	}
-	if err := verifyRecomputedStats(store, rng.Desc(), ms); err != nil {
+	if err := verifyRecomputedStats(snap, rng.Desc(), ms); err != nil {
 		t.Fatalf("failed to verify range's stats before split: %v", err)
 	}
 
@@ -432,12 +434,14 @@ func TestStoreRangeSplitStats(t *testing.T) {
 		t.Fatal(pErr)
 	}
 
+	snap = store.Engine().NewSnapshot()
+	defer snap.Close()
 	var msLeft, msRight engine.MVCCStats
-	if err := engine.MVCCGetRangeStats(store.Engine(), rng.RangeID, &msLeft); err != nil {
+	if err := engine.MVCCGetRangeStats(snap, rng.RangeID, &msLeft); err != nil {
 		t.Fatal(err)
 	}
 	rngRight := store.LookupReplica(midKey, nil)
-	if err := engine.MVCCGetRangeStats(store.Engine(), rngRight.RangeID, &msRight); err != nil {
+	if err := engine.MVCCGetRangeStats(snap, rngRight.RangeID, &msRight); err != nil {
 		t.Fatal(err)
 	}
 
@@ -458,10 +462,10 @@ func TestStoreRangeSplitStats(t *testing.T) {
 	}
 
 	// Stats should agree with recomputation.
-	if err := verifyRecomputedStats(store, rng.Desc(), msLeft); err != nil {
+	if err := verifyRecomputedStats(snap, rng.Desc(), msLeft); err != nil {
 		t.Fatalf("failed to verify left range's stats after split: %v", err)
 	}
-	if err := verifyRecomputedStats(store, rngRight.Desc(), msRight); err != nil {
+	if err := verifyRecomputedStats(snap, rngRight.Desc(), msRight); err != nil {
 		t.Fatalf("failed to verify right range's stats after split: %v", err)
 	}
 }

--- a/storage/client_test.go
+++ b/storage/client_test.go
@@ -914,9 +914,8 @@ func verifyRangeStats(eng engine.Engine, rangeID roachpb.RangeID, expMS engine.M
 	return nil
 }
 
-func verifyRecomputedStats(store *storage.Store, d *roachpb.RangeDescriptor, expMS engine.MVCCStats) error {
-	now := store.Clock().Timestamp()
-	if ms, err := storage.ComputeStatsForRange(d, store.Engine(), now.WallTime); err != nil {
+func verifyRecomputedStats(eng engine.Engine, d *roachpb.RangeDescriptor, expMS engine.MVCCStats) error {
+	if ms, err := storage.ComputeStatsForRange(d, eng, 0); err != nil {
 		return err
 	} else if expMS != ms {
 		return fmt.Errorf("expected range's stats to agree with recomputation: got\n%+v\nrecomputed\n%+v", expMS, ms)

